src/AvaloniaEdit.TextMate/TextMate.cs

```diff
@@ -61,17 +70,93 @@ public void SetGrammar(string scopeName)
 
                 _editor.TextArea.TextView.Redraw();
             }
+            
+            public bool ApplyBrushAction(string colorKeyNameFromJson, Action<IBrush> applyColorAction)
+            {
+                if (!_guiColorDictionary.TryGetValue(colorKeyNameFromJson, out var colorString))
+                {
+                    return false;
+                }
+                var colorBrush = new SolidColorBrush(Color.Parse(colorString));
+                applyColorAction(colorBrush);
+                return true;
+            }
 
-            public void SetTheme(IRawTheme theme)
+            public void SetTheme(IRawTheme theme, bool applyWindowProperties = true)
             {
                 _textMateRegistry.SetTheme(theme);
 
-                GetOrCreateTransformer().SetTheme(_textMateRegistry.GetTheme());
+                var registryTheme = _textMateRegistry.GetTheme();
+                GetOrCreateTransformer().SetTheme(registryTheme);
 
                 _tmModel?.InvalidateLine(0);
 
                 _editorModel?.InvalidateViewPortLines();
+
+                if (Application.Current?.ApplicationLifetime is IClassicDesktopStyleApplicationLifetime
+                    {
+                        MainWindow: Window window
+                    })
+                {
+                    _guiColorDictionary = registryTheme.GetGuiColorDictionary();
+                    var themeName = theme.GetName();
+
+                    // Applies to the application on a whole.. Some might want to opt out of this for their own setups.
+                    if (applyWindowProperties)
```

Comment:
I am not sure if I understand these changes, but why would control change global theme?
If text editor has custom theming solution, it should be contained by this control.

Reply:
removed..